samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImpl.java

```diff
@@ -415,6 +419,42 @@ final long getOutputWatermark() {
     }
   }
 
+  /**
+   * Returns a registry which allows registering arbitrary system-clock timer with K-typed key.
+   * The user-defined function in the operator spec needs to implement {@link TimerFunction#onTimer(Object, long)}
+   * for timer notifications.
+   * @param <K> key type for the timer.
+   * @return an instance of {@link TimerRegistry}
+   */
+  <K> TimerRegistry<K> createTimerRegistry() {
```

Comment:
Rename to createOperatorTimerRegistry to avoid confusion with TimerRegistry with for the Task.